packages/react/src/components/Copy/Copy.js

```diff
@@ -6,91 +6,101 @@
  */
 
 import PropTypes from 'prop-types';
-import React, { Component } from 'react';
+import React, { useState, useEffect, useCallback } from 'react';
+import debounce from 'lodash/debounce';
 import classnames from 'classnames';
 import { settings } from 'carbon-components';
 
 const { prefix } = settings;
 
-export default class Copy extends Component {
-  static propTypes = {
-    /**
-     * Pass in content to be rendred in the underlying <button>
-     */
-    children: PropTypes.node,
-
-    /**
-     * Specify an optional className to be applied to the underlying <button>
-     */
-    className: PropTypes.string,
-
-    /**
-     * Specify the string that is displayed when the button is clicked and the
-     * content is copied
-     */
-    feedback: PropTypes.string,
-
-    /**
-     * Specify the time it takes for the feedback message to timeout
-     */
-    feedbackTimeout: PropTypes.number,
-
-    /**
-     * Specify an optional `onClick` handler that is called when the underlying
-     * <button> is clicked
-     */
-    onClick: PropTypes.func,
-  };
-
-  static defaultProps = {
-    feedback: 'Copied!',
-    feedbackTimeout: 2000,
-    onClick: () => {},
+export default function Copy({
+  children,
+  className,
+  feedback,
+  feedbackTimeout,
+  onClick,
+  ...other
+}) {
+  const [animation, setAnimation] = useState('');
+  const classNames = classnames(className, {
+    [`${prefix}--copy-btn--animating`]: animation,
+    [`${prefix}--copy-btn--${animation}`]: animation,
+  });
+  const handleFadeOut = useCallback(
+    debounce(() => {
+      setAnimation('fade-out');
+    }, feedbackTimeout),
+    [feedbackTimeout]
+  );
+  const handleClick = useCallback(
+    event => {
+      setAnimation('fade-in');
+      onClick(event);
+      handleFadeOut();
+    },
+    [onClick, handleFadeOut]
+  );
+  const handleAnimationEnd = event => {
+    if (event.animationName === 'hide-feedback') {
+      setAnimation('');
+    }
+    if (other.handleAnimationEnd) {
+      other.handleAnimationEnd(event);
+    }
   };
 
-  state = {
-    showFeedback: false,
-  };
+  useEffect(
+    () => () => {
+      handleFadeOut.cancel();
+    },
+    [handleFadeOut]
+  );
 
-  /* istanbul ignore next */
-  componentWillUnmount() {
-    if (typeof this.timeoutId !== 'undefined') {
-      clearTimeout(this.timeoutId);
-      delete this.timeoutId;
-    }
-  }
+  return (
+    <button
+      type="button"
+      className={classNames}
+      onClick={handleClick}
```

Comment:
One thing we may want to start doing is using `composeEventHandlers` instead of proxying the call to `onClick` in the handler so that the consumer could prevent the event from firing if they wanted to.

```suggestion
      onClick={composeEventHandlers([onClick, handleClick])}
```

Reply:
in this case wouldn't preventing the event handler interfere with the desired component UX?

Reply:
It could, but would assume that the consumer understands this and is intercepting it and preventing it intentionally. Let me know if that is a bad assumption here